contracts/SimpleSwapFactory.sol

```diff
@@ -14,14 +14,20 @@ contract SimpleSwapFactory {
   /* mapping to keep track of which contracts were deployed by this factory */
   mapping (address => bool) public deployedContracts;
 
+  /* address of the ERC20-token, to be used by the to-be-deployed chequebooks.
+  If 0, deploy SimpleSwap, otherwise ERC20SimpleSwap */
+  address public ERC20Address;
+  constructor(address _ERC20Address) public {
+    ERC20Address = _ERC20Address;
+  }
   /**
   @notice deployes a new SimpleSwap contract
   @param issuer the issuer of cheques for the new chequebook
   @param defaultHardDepositTimeoutDuration duration in seconds which by default will be used to reduce hardDeposit allocations
   */
-  function deploySimpleSwap(address payable issuer, uint defaultHardDepositTimeoutDuration)
-  public payable returns (address) {
-    address contractAddress = address((new SimpleSwap).value(msg.value)(issuer, defaultHardDepositTimeoutDuration));
+  function deploySimpleSwap(address issuer, uint defaultHardDepositTimeoutDuration)
+  public returns (address) {
+    address contractAddress = address((new ERC20SimpleSwap)(issuer, ERC20Address, defaultHardDepositTimeoutDuration));
```

Comment:
Since we no longer use the `.value` call modifier you can drop the `()` around `(new ERC20SimpleSwap)`.

Reply:
done